dmt/main/models.py

```diff
@@ -465,6 +467,10 @@ def upcoming_milestone(self):
         # ideally, we want a milestone that is open, in the future,
         # and as close to today as possible
 
+        if not self.milestone_set.all().exists():
```

Comment:
The `.all()` is unnecessary here.. we can do `if not self.milestone_set.exists()`